jfr-daemon/src/main/java/com/newrelic/jfr/agent/AgentController.java

```diff
@@ -0,0 +1,96 @@
+package com.newrelic.jfr.agent;
+
+import com.newrelic.jfr.daemon.DaemonConfig;
+import com.newrelic.jfr.daemon.JFRUploader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.ParseException;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import javax.management.*;
+import jdk.jfr.Configuration;
+import jdk.jfr.Recording;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AgentController {
```

Comment:
The visibility of this class and all its methods can be reduced to package private.

Reply:
Done.